spi/src/main/java/org/mapstruct/spi/AccessorNamingStrategy.java

```diff
@@ -0,0 +1,134 @@
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be an adder, {@code false} otherwise.
+     */
+    boolean isAdderName(String methodName);
+
+    /**
+     * Extract the property name from a method name for which {@link #isNonBooleanGetterName(String)} returned
+     * {@code true}.
+     * <p>
+     * Default implementation: Remove first 3 characters ("get") of the method name, and
+     * {@link java.beans.Introspector#decapitalize(String) decapitalize} the result.
+     *
+     * @param methodName The method name, guaranteed to be a non-boolean getter name.
+     * @return The property name corresponding to the method name.
+     */
+    String getPropertyNameForNonBooleanGetterName(String methodName);
+
+    /**
+     * Extract the property name from a method name for which {@link #isBooleanGetterName(String)} returned
+     * {@code true}.
+     * <p>
+     * Default implementation: Remove the first 2 characters ("is") of the method name, and
+     * {@link java.beans.Introspector#decapitalize(String) decapitalize} the result.
+     *
+     * @param methodName The method name, guaranteed to be a boolean getter name.
+     * @return The property name corresponding to the method name.
+     */
+    String getPropertyNameForBooleanGetterName(String methodName);
+
+    /**
+     * Extract the property name from a method name for which {@link #isSetterName(String)} returned {@code true}.
+     * <p>
+     * Default implementation: Remove the first 3 characters ("set") of the method name, and
+     * {@link java.beans.Introspector#decapitalize(String) decapitalize} the result.
+     *
+     * @param methodName The method name, guaranteed to be a setter name.
+     * @return The property name corresponding to the method name.
+     */
+    String getPropertyNameForSetterName(String methodName);
+
+    /**
+     * Extract the element name (singular form of the collection's property name) from a method name for which
+     * {@link #isAdderName(String)} returned {@code true}.
+     * <p>
+     * Default implementation: Remove the first 3 characters ("add") of the method name, and
+     * {@link java.beans.Introspector#decapitalize(String) decapitalize} the result.
+     *
+     * @param methodName The method name, guaranteed to be an adder name.
+     * @return The element name corresponding to the method name.
+     */
+    String getElementNameForAdderName(String methodName);
+
+    /**
+     * Extract the non-boolean getter method name from the setter method name of the same property.
+     * <p>
+     * Default implementation: Replace the first 3 characters ("get") of the method name with "set".
+     *
+     * @param methodName The method name, guaranteed to be a setter name.
+     * @return The corresponding non-boolean getter method name.
+     */
+    String getNonBooleanGetterNameForSetterName(String methodName);
```

Comment:
To sum up, WDYT about having just the following methods on this contract:

* `isPropertySetter(String method)`
* `isPropertyGetter(String method)`
* `isAdder(String method)`
* `getPropertyName(String getterOrSetterMethod)`
* `getElementName(String adderMethod)`
* `getGetterName(String property)`

?

This seems a bit simpler on implementors to me. You could even think about collapsing the first three into a single one `MethodType getType(String method)` returning an enum with values `SETTER`, `GETTER`, `ADDER` and `OTHER`?

Any thoughts?